modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/DeployService.java

```diff
@@ -709,4 +657,43 @@ public List<ServiceVo> listDeployedServicesOfIsv(ServiceQueryModel query) {
                         .equals(deployServiceEntity.getNamespace()))
                 .map(this::convertToServiceVo).toList();
     }
+
+    /**
+     * Get deploy service detail by id.
+     *
+     * @param id ID of deploy service.
+     * @return serviceDetailVo
+     */
+    public ServiceDetailVo getServiceDetailsByIdForIsv(UUID id) {
+        DeployServiceEntity deployServiceEntity = getDeployServiceEntity(id);
+        ServiceHostingType serviceHostingType =
+                deployServiceEntity.getDeployRequest().getServiceHostingType();
+        if (ServiceHostingType.SERVICE_VENDOR == serviceHostingType) {
```

Comment:
the check should be other way round. if hosting is not equal to SERVICE_VENDOR then reject the request.